components/p3a/nitro_utils/attestation.cc

```diff
@@ -266,4 +303,25 @@ void RequestAndVerifyAttestationDocument(
       kAttestationBodyMaxSize);
 }
 
+bool VerifyNonceForTesting(const std::vector<uint8_t>& attestation_bytes,
+                           const std::vector<uint8_t>& orig_nonce) {
+  auto document = cbor::Reader::Read(attestation_bytes);
+  if (!document.has_value() || !document.value().is_map()) {
+    LOG(ERROR) << "Nitro verification: expected cbor map for test";
+    return false;
+  }
+  return VerifyNonce(document.value().GetMap(), orig_nonce);
+}
+
+bool VerifyUserDataKeyForTesting(
+    const std::vector<uint8_t>& attestation_bytes,
+    scoped_refptr<net::X509Certificate> expected_cert) {
+  auto document = cbor::Reader::Read(attestation_bytes);
```

Comment:
it seems that you could just have a helper method in `NitroAttestationTest` that handles the parsing of the cbor, and then you could pass the result directly to `VerifyUserDataKey` and `VerifyNonce`

Reply:
I was trying to follow the style guide about [testing-specific code](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md#tests-and-test_only-code) to maintain [internal linkage](https://google.github.io/styleguide/cppguide.html#Internal_Linkage) for the actual verification functions.

Seems a bit clunky either way. You prefer exposing them?

Reply:
ah sorry, for some reason I thought that the two functions were already exposed. I think this looks good as is. Maybe it would be good to move the cbor reading code to a function in the anonymous namespace to make the code more DRY, but up to you.

Reply:
Ok, thanks for clarifying.

It's only a couple of lines, and we'd still have to check the return code, so I'll leave the duplication. If there were three copies, it would be worth a separate function.